File: conversor_moedas_java/app/src/main/java/com/example/conversor_moedas_java/MainActivity.java
package com.example.conversor_moedas_java;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private ViewHolder mViewHolder = new ViewHolder();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        this.mViewHolder.edit_value = findViewById(R.id.edit_value);
    }

    @Override
    protected void onResume() {
        super.onResume();
        Toast.makeText(this,"Resume executado",Toast.LENGTH_SHORT).show();
    }

    private static class ViewHolder{
        //Classe estática que conterá todos os elementos de interface, que serão instanciados
        // apenas uma vez no onCreate

        EditText edit_value;
    }
}

File: AboutMe/app/src/main/java/com/example/aboutme/MainActivity.kt
package com.example.aboutme

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.databinding.DataBindingUtil
import com.example.aboutme.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private val myName: MyName = MyName("Amora - a Labradora")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_main)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
/*
        findViewById<Button>(R.id.button_done).setOnClickListener{
            addNickname(it)
        }
  */
        binding.myName = myName
        binding.buttonDone.setOnClickListener {
            addNickname(it)
        }
    }

    private fun addNickname(view: View) {
        /*
        val edittext_nickname = findViewById<EditText>(R.id.edittext_nickname)
        val textview_nickname = findViewById<TextView>(R.id.text_nickname)
*/
        binding.apply {
            myName?.nickname = nicknameEdit.text.toString()
            //binding.nicknameText.text = binding.nicknameEdit.text
            invalidateAll()
            binding.nicknameEdit.visibility = View.GONE
            binding.buttonDone.visibility = View.GONE
            binding.nicknameText.visibility = View.VISIBLE
        }
    }

}